core/src/Cabana_SiloParticleOutput.hpp

```diff
@@ -86,17 +86,18 @@ namespace Impl
 template <class SliceType>
 void writeFields(
     DBfile* silo_file, const std::string& mesh_name, const SliceType& slice,
+    const std::size_t num_particle,
     typename std::enable_if<
         2 == SliceType::kokkos_view::traits::dimension::rank, int*>::type = 0 )
 {
     // Reorder in a contiguous blocked format.
     Kokkos::View<typename SliceType::value_type*,
                  typename SliceType::device_type>
-        view( "field", slice.size() );
+        view( "field", num_particle );
     Kokkos::parallel_for(
         "Cabana::SiloParticleOutput::writeFieldRank0",
         Kokkos::RangePolicy<typename SliceType::execution_space>(
-            0, slice.size() ),
+            0, num_particle ),
```

Comment:
When there is 10 particle, for example, we can only write from `0` to `6`? Can we also write from `2` to `8`?

Reply:
Good point - added your suggestion